cloudinit/config/cc_set_passwords.py

```diff
@@ -172,11 +194,18 @@ def handle(_name, cfg, cloud, log, args):
         password = util.get_cfg_option_str(cfg, "password", None)
 
     expire = True
-    plist = None
+    plist: List = []
+    users_list: List = []
 
     if "chpasswd" in cfg:
         chfg = cfg["chpasswd"]
+        users_list = util.get_cfg_option_list(chfg, "users")
         if "list" in chfg and chfg["list"]:
+            log.warning(
+                "DEPRECATION: The list key is "
+                "deprecated and will be removed from a future version of "
+                "cloud-init. Use the users key instead."
```

Comment:
We should standardize on the deprecation format we've used elsewhere in other cc_* modules
```suggestion
                "DEPRECATION: key 'lists' is now deprecated. Use 'users'."
```